frontend/js/src/playlists/components/ImportPlaylistModal.tsx

```diff
@@ -0,0 +1,204 @@
+import * as React from "react";
+import NiceModal, { useModal } from "@ebay/nice-modal-react";
+import { toast } from "react-toastify";
+
+import GlobalAppContext from "../../utils/GlobalAppContext";
+import { ToastMsg } from "../../notifications/Notifications";
+
+type ImportPlaylistModalProps = {
+  playlist?: JSPFPlaylist;
+  initialTracks?: JSPFTrack[];
+};
+
+export default NiceModal.create((props: ImportPlaylistModalProps) => {
+  const modal = useModal();
+  const closeModal = React.useCallback(() => {
+    modal.hide();
+    setTimeout(modal.remove, 3000);
+  }, [modal]);
+
+  const { currentUser, APIService } = React.useContext(GlobalAppContext);
+  const { playlist, initialTracks } = props;
+
+  const [fileError, setFileError] = React.useState<string | null>(null);
+  const [newPlaylist, setNewPlaylist] = React.useState<JSPFObject | null>(null);
```

Comment:
I think the names here are a bit confusing, it took me a while to understand that you're not setting the newly created playlist in the state, but rather the content of the uploaded file if I understood correctly.
Maybe `fileContent, setFileContent` or something like that?